### ISSUE TITLE
Forum disappears / yoast seo 9.7 compatibility issue

### ISSUE THREAD

**Issue by chaosversum** [Asgaros/asgaros-forum#207]
My Forum starts to disappear after the recent update from 1.13.0 to 1.13.2

First load of the forum page after the update works well, but after a reload of the page, the forum disappears from the page (the part rendered by the [forum] shortcode). It keeps disappeared until i save something in the backend, then it shows up again, but only one time until i save something again.

I tracked the issue down to the forum-compatibility.php File line 42. Removing the line "remove_action('wp_head', array($wpseo_front, 'head'), 1);" fixed the problem.

I'm using the newest Yoast SEO version 9.7, so it seams to be a compatibility issue with that recent version.

**Comment by Asgaros** [comment]
Hello @chaosversum 

I tested it with the latest version of Yoast SEO 9.7 and cannot reproduce the error. Are you sure that it is really caused by the interaction between Asgaros Forum and Yoast? Please also try to disable other plugins one by one - for me the behavior sounds more like an issue with some caching-plugin.

**Comment by chaosversum** [comment]
I digged deeper into the problem and found something:

Yoast SEO calls wp_reset_query in the WPSEO_Frontend->head() function, which sets the global $post variable. By removing that function in the forum-compatibility.php, the global $post variable is no longer set "automatically" and the "the_content()" function call in our custom forum page template does not work anymore.

Since we are using yoast seo on all of our projects, we are not used to using the proper wordpress loop in our page templates. I fixed the issue by calling the_post() before calling the_content() in our custom forum page template.

Thanks for testing it out on your side and sorry for the confusion i may have caused.
I will close the issue. Maybe it can help someone else with a similar problem, at least.
